### PR TITLE
K8SPXC-264 Remove EKS related pod fields

### PR DESCRIPTION
**Pull request by cap1984**

### CODE REVIEW

**Comment by cap1984** [comment]
trying to avoid this
```
+ diff -u /mnt/jenkins/workspace/pxc-operator-eks/source/e2e-tests/security-context/compare/pod_restore-src-restore-pvc-sec-context.yml /tmp/tmp.gazVqwibnP/pod_restore-src-restore-pvc-sec-context.yml
--- /mnt/jenkins/workspace/pxc-operator-eks/source/e2e-tests/security-context/compare/pod_restore-src-restore-pvc-sec-context.yml	2020-04-23 09:43:28.230852586 +0000
+++ /tmp/tmp.gazVqwibnP/pod_restore-src-restore-pvc-sec-context.yml	2020-04-23 10:11:38.844618287 +0000
@@ -1,6 +1,8 @@
 apiVersion: v1
 kind: Pod
 metadata:
+  annotations:
+    kubernetes.io/psp: eks.privileged
   deletionGracePeriodSeconds: 30
   labels:
     name: restore-src-restore-pvc-sec-context

```

File: e2e-tests/functions [hunk 1]
```diff
@@ -227,6 +227,11 @@ compare_kubectl() {
         | $sed -e '/name: S3_BUCKET_PATH/,+1d' \
         | $sed -e '/name: S3_BUCKET_URL/,+1d' \
         > ${new_result}
+
+    # Trying to get rid extra fields on EKS
+    if [[ "${resource}" == pod* ]]; then
+        $sed -i -E '/annotations:|kubernetes.io\/psp:/d' ${new_result}
+    fi
```
Comment (delgod) [inline e2e-tests/functions:234]:
```suggestion
        | $sed -e '/name: S3_BUCKET_URL/,+1d' \
        | $sed -e '/annotations:$/{$!{N;s/annotations:\n    kubernetes.io\/psp://;d}}' \
        > ${new_result}
```

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | failed |
| scaling | failed |
| recreate | passed |
| storage | failed |
| self-healing | failed |
| scaling-proxysql | failed |
| big-data | failed |
| upgrade | passed |
| restore-to-encrypted-cluster | failed |
| upgrade-consistency | failed |
| limits | failed |
| monitoring | failed |
| security-context | failed |
| self-healing-advanced | passed |
| demand-backup | failed |
| monitoring-2-0 | failed |
| affinity | failed |
| one-pod | passed |
| operator-self-healing | passed |
| scheduled-backup | failed |
| auto-tuning | failed |
| demand-backup-encrypted-with-tls | passed |
| proxysql-sidecar-res-limits | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| init-deploy | failed |
| recreate | failed |
| storage | passed |
| big-data | failed |
| scaling-proxysql | passed |
| self-healing | failed |
| self-healing-advanced | failed |
| upgrade | passed |
| limits | passed |
| restore-to-encrypted-cluster | passed |
| upgrade-consistency | passed |
| security-context | passed |
| operator-self-healing | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| affinity | passed |
| one-pod | passed |
| scheduled-backup | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | failed |
| demand-backup-encrypted-with-tls | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| big-data | failed |
| scaling | passed |
| storage | passed |
| init-deploy | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| restore-to-encrypted-cluster | passed |
| upgrade | passed |
| limits | passed |
| self-healing-advanced | failed |
| upgrade-consistency | passed |
| security-context | passed |
| monitoring | failed |
| demand-backup | passed |
| operator-self-healing | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| scheduled-backup | failed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| demand-backup-encrypted-with-tls | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| scaling-proxysql | passed |
| upgrade | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| limits | passed |
| security-context | passed |
| monitoring | passed |
| storage | passed |
| self-healing | passed |
| self-healing-advanced | failed |
| big-data | passed |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| scheduled-backup | failed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| operator-self-healing | passed |
| demand-backup-encrypted-with-tls | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| scaling-proxysql | passed |
| upgrade | passed |
| recreate | passed |
| upgrade-consistency | passed |
| security-context | passed |
| restore-to-encrypted-cluster | passed |
| big-data | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| storage | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| init-deploy | passed |
| limits | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| operator-self-healing | passed |
| demand-backup-encrypted-with-tls | failed |

**Comment by JNKPercona** [comment]
PXC operator docker - perconalab/percona-xtradb-cluster-operator:PR-390-8d6cf3be

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| scaling-proxysql | passed |
| upgrade | passed |
| init-deploy | passed |
| recreate | passed |
| upgrade-consistency | passed |
| limits | passed |
| restore-to-encrypted-cluster | passed |
| security-context | passed |
| monitoring | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| scheduled-backup | passed |
| big-data | passed |
| affinity | passed |
| demand-backup-encrypted-with-tls | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| storage | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |

**Pull request merged**